### ISSUE TITLE
Cannot retrieve full list of rooms

### ISSUE THREAD

**Issue by Fishbowler** [igniterealtime/openfire-restAPI-plugin#118]
Only applies in main following the fix in #114 
When there is no search term provided, the list never gets populated, and an empty list of rooms is returned.

**Comment by guusdk** [comment]
Exactly how do you reproduce the issue? This seems to work for me with the current 'main' branch:

```
curl -X 'GET' \
  'http://localhost:9090/plugins/restapi/v1/chatrooms?servicename=conference&type=all&expandGroups=false' \
  -H 'accept: application/xml' \
  -H 'Authorization: test'
```

```xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
  <chatRooms>
    <chatRoom>
      <roomName>room5</roomName>
      <naturalName>room5</naturalName>
      <description>room5</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.766+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.779+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room6</roomName>
      <naturalName>room6</naturalName>
      <description>room6</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.787+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.838+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room3</roomName>
      <naturalName>room3</naturalName>
      <description>room3</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.696+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.704+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room4</roomName>
      <naturalName>room4</naturalName>
      <description>room4</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.708+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.759+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room9</roomName>
      <naturalName>room9</naturalName>
      <description>room9</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.909+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.914+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room7</roomName>
      <naturalName>room7</naturalName>
      <description>room7</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.843+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.850+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room8</roomName>
      <naturalName>room8</naturalName>
      <description>room8</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.854+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.906+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room1</roomName>
      <naturalName>room1</naturalName>
      <description>room1</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.618+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.638+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room2</roomName>
      <naturalName>room2</naturalName>
      <description>room2</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.644+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.691+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room10</roomName>
      <naturalName>room10</naturalName>
      <description>room10</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.918+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.966+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room11</roomName>
      <naturalName>room11</naturalName>
      <description>room11</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.970+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.974+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room12</roomName>
      <naturalName>room12</naturalName>
      <description>room12</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.978+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:18.983+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
    <chatRoom>
      <roomName>room13</roomName>
      <naturalName>room13</naturalName>
      <description>room13</description>
      <subject>
      </subject>
      <creationDate>2022-07-28T08:59:18.987+02:00</creationDate>
      <modificationDate>2022-07-28T08:59:19.033+02:00</modificationDate>
      <maxUsers>0</maxUsers>
      <persistent>true</persistent>
      <publicRoom>true</publicRoom>
      <registrationEnabled>true</registrationEnabled>
      <canAnyoneDiscoverJID>true</canAnyoneDiscoverJID>
      <canOccupantsChangeSubject>false</canOccupantsChangeSubject>
      <canOccupantsInvite>false</canOccupantsInvite>
      <canChangeNickname>true</canChangeNickname>
      <logEnabled>true</logEnabled>
      <loginRestrictedToNickname>false</loginRestrictedToNickname>
      <membersOnly>false</membersOnly>
      <moderated>false</moderated>
      <broadcastPresenceRoles>
        <broadcastPresenceRole>moderator</broadcastPresenceRole>
        <broadcastPresenceRole>participant</broadcastPresenceRole>
        <broadcastPresenceRole>visitor</broadcastPresenceRole>
      </broadcastPresenceRoles>
      <owners>
        <owner>jane@example.org</owner>
      </owners>
      <admins/>
      <members/>
      <outcasts/>
      <ownerGroups/>
      <adminGroups/>
      <memberGroups/>
      <outcastGroups/>
    </chatRoom>
  </chatRooms>
```

**Comment by Fishbowler** [comment]
I've failed to properly tag the fix with a #.
This should've been automatically closed when the fix went in via #119.
